evaluation.md

```diff
@@ -6,14 +6,136 @@ _Describe each change from your ideal syntax to the syntax you implemented, and
 describe_ why _you made the change._
 
 **On a scale of 1–10 (where 10 is "a lot"), how much did you have to change your syntax?**
+The following are some observations we made and challenges we encountered when creating our internal DSL.
+
+* Initially, we thought it would be very easy to move sequentially from "command" to the next (each line of instructions is a command). However, we then realized that it's tricky to have some commands include an explicit designation of the next command to jump to, and for other commands to not include that explicit designation. At first, we didn't want to write " +1 " at the end of every command if we wanted the machine to proceed by evaluating the next step. We only wanted to specify the next step if we wanted the machine to jump to any command NOT directly after the current command, check if the condition of the command is satisfied, move in the designated direction if so, and check the condition of the next command if the condition of the current command is not satisfied. We also wanted to use the asterisk to instruct the machine to reevaluate the condition of the current command (the condition being the combination of free/occupied States) again, and move accordingly, until the condition of that command is no logner satisfied.. However, we didn't know how to keep track of a counter that knew the current state, so we had to explicitly name the CurrentState and the NextState.  
```

Comment:
This was also a pain for us when implementing our internal DSL.  Having to specify states was annoying since we wanted to do things based on directions.